src/lib.rs

```diff
@@ -13,12 +13,25 @@
 #![warn(rust_2018_idioms)]
 #![cfg_attr(feature = "nightly", feature(asm))]
 
+cfg_if::cfg_if! {
+    if #[cfg(target_arch = "wasm32")] {
```

Comment:
This wont work correctly when using the multi threading wasm extension.

Reply:
This is true, but neither would the original code without being explicitly designed for the new atomic primitives that aren’t even available on nightly, right? If it needs new code to work I’m not fussed about the exact way it happens to be incorrect now.

Reply:
Maybe add

```rust
#[cfg(all(target_arch = "wasm32", target_feature = "atomics"))]
compile_error!("parking_lot doesnt yet support multi threaded wasm");
```

Also atomic.notify and atomic.wait are exposed on nightly: https://doc.rust-lang.org/stable/core/arch/wasm32/index.html Normal atomics are accessable using the Atomic* types.

Reply:
I was referring mainly to having to compile libstd from source. Regardless, this is probably a good idea.